File: Exerc5/MidHandler/src/message/ReplyHeader.java
package message;

import java.io.Serializable;

public class ReplyHeader implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -2342544144018128812L;
	private String serviceContext;
	private int requestId;
	private int replyStatus;
	
	/**
	 * @param serviceContext
	 * @param requestId
	 * @param replyStatus
	 */
	public ReplyHeader(String serviceContext, int requestId, int replyStatus) {
		super();
		this.serviceContext = serviceContext;
		this.requestId = requestId;
		this.replyStatus = replyStatus;
	}
	/**
	 * @return the serviceContext
	 */
	public String getServiceContext() {
		return serviceContext;
	}
	/**
	 * @param serviceContext the serviceContext to set
	 */
	public void setServiceContext(String serviceContext) {
		this.serviceContext = serviceContext;
	}
	/**
	 * @return the requestId
	 */
	public int getRequestId() {
		return requestId;
	}
	/**
	 * @param requestId the requestId to set
	 */
	public void setRequestId(int requestId) {
		this.requestId = requestId;
	}
	/**
	 * @return the replyStatus
	 */
	public int getReplyStatus() {
		return replyStatus;
	}
	/**
	 * @param replyStatus the replyStatus to set
	 */
	public void setReplyStatus(int replyStatus) {
		this.replyStatus = replyStatus;
	}
}

File: Projeto/PythonServer/message/ReplyHeader.py
import message_pb2

class ReplyHeader(object):
    #--------------Constructor--------------
    def __init__(self,serviceContext,requestId,replyStatus):
        self.__replyHeader = message_pb2.ReplyHeader()
        self.__replyHeader.serviceContext = serviceContext
        self.__replyHeader.requestId = requestId
        self.__replyHeader.replyStatus = replyStatus

    #---------------------------------------

    #----------Getters and Setters----------
    def getServiceContext(self):
        return self.__replyHeader.serviceContext

    def getRequestId(self):
        return self.__replyHeader.requestId

    def getReplyStatus(self):
        return self.__replyHeader.replyStatus

    def setServiceContext(self,serviceContext):
        self.__replyHeader.serviceContext = serviceContext

    def setRequestId(self,requestId):
        self.__replyHeader.requestId = requestId

    def setReplyStatus(self,replyStatus):
        self.__replyHeader.replyStatus = replyStatus

    #---------------------------------------